orders/models.py

```diff
@@ -71,3 +72,56 @@ def cancel_order(self):
 
     def get_order_total_price(self):
         return self.price_per_day * (self.end_date - self.start_date).days
+
+    def get_order_status(self):
+        if self.status == StatusOptions.Pending:
+            return "Pending"
+        elif self.status == StatusOptions.Approved:
+            return "Approved"
+        elif self.status == StatusOptions.Canceled:
+            return "Canceled"
+        elif self.status == StatusOptions.OnGoing:
+            return "On going"
+        return "Finished"
+
+    @staticmethod
+    def get_capacity_of_daycare_in_dates_range(daycare_id, start_date, end_date):
+        capacity_per_day_list = [0] * ((end_date - start_date).days + 2)
+        relevant_orders = Order.objects.filter(daycare_id=daycare_id, status__in=['A', 'O'])
+        start_date = datetime.date(year=start_date.year, month=start_date.month, day=start_date.day)
+        end_date = datetime.date(year=end_date.year, month=end_date.month, day=end_date.day)
+
+        for order in relevant_orders:
+            if end_date < order.start_date or order.end_date < start_date:
```

Comment:
Can you explain what's the logic here?

Reply:
Let's give the orders names for better explanation:
1. Checked order - the order we want to check that can be approved. 
2. Current order - the order from the list of relevant_orders.

We want to check the dates that the current order take place, because if the orders are not having shared dates, then the check for capacity on those dates is irrelevant, and we want to skip that.
So, if the current order is ending before the checked order, or the checked order is ending before the current order, then we can skip to the next order from the list.

Reply:
I'm pretty sure you can pull specific dates from the DB instead of dealing with this complicated logic.

Reply:
Added ranges of orders to the filter.